docs/guides/biolatency.md

```diff
@@ -0,0 +1,130 @@
+distribution of I/O latency (time), to then print this as a histogram when
+the gadget is stopped. Notice that the latency of the disk I/O is measured
+from the issue to the device to its completion, so it does not include
+time queued in the kernel. It means that the histogram reflects only the
+performance of the device and not the latency suffered by an application.
+
+The histogram shows the number of I/O operations (`count` column) that lie in
+the latency range `interval-start` -> `interval-end` (`usecs` column), which,
+as the columns name indicates, is given in microseconds.
+
+For this guide, we will use
+[the `stress` tool](https://linux.die.net/man/1/stress) that allows us to load
+and stress the system in many different ways. In particular, we will use
+the `--io` flag that will generate a given number of workers to spin on the
+[sync() syscall](https://man7.org/linux/man-pages/man2/sync.2.html). In this
+way, we will generate disk I/O that we will analyse using the biolatency
+gadget.
+
+Firstly, let's use the biolatency gadget to see the I/O latency in our
+testing node with its normal load work:
+
+```bash
+# Start the gadget on the worker-node node
+$ kubectl gadget biolatency start --node worker-node
+4b5501BrEjiw2GxG
+
+# Wait for around 3 minutes
+
+# Stop the gadget to generate the histogram
+$ kubectl gadget biolatency stop 4b5501BrEjiw2GxG
+Tracing block device I/O... Hit Ctrl-C to end.
+
+     usecs               : count     distribution
+         0 -> 1          : 0        |                                        |
+         2 -> 3          : 0        |                                        |
+         4 -> 7          : 0        |                                        |
+         8 -> 15         : 0        |                                        |
+        16 -> 31         : 0        |                                        |
+        32 -> 63         : 17       |*                                       |
+        64 -> 127        : 261      |*******************                     |
+       128 -> 255        : 546      |****************************************|
+       256 -> 511        : 426      |*******************************         |
+       512 -> 1023       : 227      |****************                        |
+      1024 -> 2047       : 18       |*                                       |
+      2048 -> 4095       : 8        |                                        |
+      4096 -> 8191       : 23       |*                                       |
+      8192 -> 16383      : 15       |*                                       |
+     16384 -> 32767      : 2        |                                        |
+     32768 -> 65535      : 1        |                                        |
+```
+
+This output shows that the bulk of the I/O was between 64 and 1023 us, and
+that there were 1544 I/O operations during the time the gadget was running.
```

Comment:
did you sum it up by hand? :thinking:

Reply:
Unfortunately yes, that's something the tool provides using the option `-e`, see [here](https://github.com/iovisor/bcc/blob/67f59ee80fcf5deedaacba1436d9fa09d32a16a0/tools/biolatency_example.txt#L296-L314) but we don't have the possibility of using flags yet. 

I think we should definitely extend the trace CR to allow sending flags to the gadgets. It would be useful also other gadgets as well, for instance, to show only UDP or TCP sockets with the `socket-collector` gadget.